packages/itwinui-react/src/core/Tile/TileNew.tsx

```diff
@@ -0,0 +1,646 @@
+            'iui-actionable': isActionable,
+            [`iui-${status}`]: !!status,
+            'iui-loading': isLoading,
+          },
+          className,
+        )}
+        aria-disabled={isDisabled}
+        ref={ref}
+        {...rest}
+      />
+    </TileContext.Provider>
+  );
+}) as PolymorphicForwardRefComponent<'div', TileOwnProps>;
+TileComponent.displayName = 'TileNew';
+
+type TileOwnProps = {
+  /**
+   * Status of the tile.
+   */
+  status?: 'positive' | 'warning' | 'negative';
+  /**
+   * Whether the tile is selected or in "active" state.
+   * Gets highlighted and shows a checkmark icon near tile name.
+   */
+  isSelected?: boolean;
+  /**
+   * Whether the tile is "new". Tile name becomes bold and gets a new status icon.
+   */
+  isNew?: boolean;
+  /**
+   * Default tile variant or the folder layout.
+   * @default 'default'
+   */
+  variant?: 'default' | 'folder';
+  /**
+   * Whether the tile is expected to be interactable (i.e. `onClick`).
+   * It becomes focusable and gets on hover styling.
+   */
+  isActionable?: boolean;
+  /**
+   * Display a loading state.
+   * @default false
+   */
+  isLoading?: boolean;
+  /**
+   * Flag whether the tile is disabled.
+   *
+   * Note: This only affects the tile. You need to manually disable
+   * the buttons and other interactive elements inside the tile.
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+  setActionable: React.Dispatch<React.SetStateAction<boolean>>;
+  onClick?: React.MouseEventHandler<HTMLElement>;
+};
+
+// ----------------------------------------------------------------------------
+// Tile.ThumbnailArea component
+
+const TileThumbnailArea = polymorphic.div('iui-tile-thumbnail');
+TileThumbnailArea.displayName = 'TileNew.ThumbnailArea';
+
+type TileThumbnailAreaOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.ThumbnailPicture component
+
+const TileThumbnailPicture = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', className, children, ...rest } = props;
+  return (
+    <Element
+      className={cx('iui-tile-thumbnail-picture', className)}
+      style={{
+        backgroundImage:
+          children && typeof children === 'string'
+            ? `url(${children})`
+            : undefined,
+      }}
+      ref={ref}
+      {...rest}
+    />
+  );
+}) as PolymorphicForwardRefComponent<'div', TileThumbnailPictureOwnProps>;
+
+TileThumbnailPicture.displayName = 'TileNew.TileThumbnailPicture';
+
+type TileThumbnailPictureOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+// ----------------------------------------------------------------------------
+// Tile.ThumbnailAvatar component
+
+const TileThumbnailAvatar = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', className, children, ...rest } = props;
+  if (React.isValidElement(children)) {
```

Comment:
I'm not exactly sure what this check is for. Do you mind leaving a comment in the code for what this check is for?

Reply:
Actually want to pull @mayank99 in on this one to ask because I copied the logic from original Tile, which is that the `thumbnail` or `children` in this case is a custom React Element for thumbnail that the user passed in, and we cloned it to render 
![image](https://github.com/iTwin/iTwinUI/assets/22462084/b798cbc2-b9e0-4273-9ab0-930148827afd)

So I think the check is for whether what is passed in `thumbnail` is a custom React Element. As opposed to being just a string to a thumbnail image.

But also, I circumvented `React.cloneElement` by using `React.createElement` in order to keep the same logic but without using `cloneElement`. Should we do it like that? Or if not, what else can we do here?
![image](https://github.com/iTwin/iTwinUI/assets/22462084/24fe51c2-f207-4380-a3ae-d357f95a02b6)

Reply:
Please avoid using deprecated apis such as `cloneElement` or `createElement`. Writing such code increases complexity and is prone to bugs, while providing little benefit.

Ultimately, what we want to achieve is pretty simple: for both `Tile.ThumbnailPicture` and `Tile.ThumbnailAvatar`, we only need to support _two_ cases. The developer should be able to pass either:
- a string containing the url to the image, or
- a component that renders an `<img>` or `<svg>` or `<div>` with background-image

You can use a [discriminated union](https://www.totaltypescript.com/discriminated-unions-are-a-devs-best-friend) type to narrow the prop types (similar to #1120). Something like:
```
{
  url: string
} |
{
  children: string
}
```

After that, you can render a regular div with your css class (whether it's `iui-tile-thumbnail-picture` or `iui-tile-thumbnail-avatar` or whatever).
```jsx
<div
  className={cx('iui-whatever', className)}
  style={{ backgroundImage: `url()`, ...style }}
  {...rest}
/>
``` 
```jsx
<div
  className={cx('iui-whatever', className)}
  {...rest}
>
  {children}
</div>
```

After that, you can handle all cases (svg vs img vs div) inside the css code.

Reply:
Oop I did not realize `createElement` is deprecated 😬 but thanks for the direction 😅

Reply:
I updated this. Working as it should at least on Storybook 🤔 also I removed the use of ThumbnailAvatar altogether and only use ThumbnailPicture to handle both cases. The user will have to use a `url` prop to pass in the link for their image, or they could pass in their own Avatar/SVG. Did not clean up Avatar code yet, trying to see if this is what we want.